### PR TITLE
fix(map-validator): Jessie object accessing enum value

### PR DESCRIPTION
**Pull request by martinalbert**
## Description

- unable to access correct value using [] - instead throws:
```
Wrong Variable Structure: variable statusMap expected enumVal1 or enumVal2, but got { 
    enumVal1: "enumVal1", 
    enumVal2: "enumVal2",
}
```

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTION_GUIDE** document.
- [ ] I haven't repeated the code. (DRY)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/interpreter/utils.ts [hunk 1]
```diff
@@ -417,7 +417,7 @@ export function findTypescriptProperty(name: string, node: ts.Node): boolean {
 }
 
 export function getTypescriptIdentifierName(node: ts.Node): string {
-  if (ts.isIdentifier(node)) {
+  if (ts.isIdentifier(node) || ts.isStringLiteral(node)) {
```
Comment (TheEdward162) [inline src/interpreter/utils.ts:420]: question: Why is this change needed?

**Review by TheEdward162 — APPROVED** [review]

File: src/interpreter/utils.ts (hunk 1 above)
Comment (martinalbert) [reply src/interpreter/utils.ts]: Added it because of another change (was passing `argumentExpression` which could be string literal), but looking at references of this function it wasn't used at all, even though it was doing the same thing as `getVariableName` function, so I moved this logic there.

**Comment by freaz** [comment]
@martinalbert I did quick test and it looks it doesn't work well. It can be bad test, but what triggered suspicions was removed recursion.

```ts
  describe('getVariableName', () => {
    it("return o.f1.f2 for o['f1']['f2']", () => {
      const jessieExpression = new ScriptExpressionCompiler("o['f1']['f2']");
      const expr = (jessieExpression.rawExpressionNode as ElementAccessExpression);
      expect(getVariableName(expr)).toBe('o.f1.f2');
    });
  });
```

image

**Comment by freaz** [comment]
@martinalbert I started with hopefully more robust solution [here](https://github.com/superfaceai/parser/commit/6963a72d0dc1d51c33e98f1cb5c58adf8cd5febf)

**Comment by freaz** [comment]
Superseded by #105
